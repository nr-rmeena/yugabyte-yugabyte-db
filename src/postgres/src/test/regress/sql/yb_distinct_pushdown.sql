CREATE TABLE distinct_pushdown_table(r1 INT, r2 INT, PRIMARY KEY(r1 ASC, r2 ASC));
INSERT INTO distinct_pushdown_table (SELECT 1, i FROM GENERATE_SERIES(1, 1000) AS i);

-- Disable DISTINCT pushdown
SET yb_enable_distinct_pushdown TO off;

-- Must pull even duplicate rows without pushdown. Verify that using EXPLAIN ANALYZE
EXPLAIN (ANALYZE, COSTS OFF, TIMING OFF, SUMMARY OFF) SELECT DISTINCT r1 FROM distinct_pushdown_table WHERE r1 <= 10;

-- Enable DISTINCT pushdown
SET yb_enable_distinct_pushdown TO on;

-- Must pull fewer rows with pushdown. Verify that using EXPLAIN ANALYZE
EXPLAIN (ANALYZE, COSTS OFF, TIMING OFF, SUMMARY OFF) SELECT DISTINCT r1 FROM distinct_pushdown_table WHERE r1 <= 10;

DROP TABLE distinct_pushdown_table;
