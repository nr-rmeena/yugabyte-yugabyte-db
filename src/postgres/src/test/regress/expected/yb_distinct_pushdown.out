CREATE TABLE distinct_pushdown_table(r1 INT, r2 INT, PRIMARY KEY(r1 ASC, r2 ASC));
INSERT INTO distinct_pushdown_table (SELECT 1, i FROM GENERATE_SERIES(1, 1000) AS i);
-- Disable DISTINCT pushdown
SET yb_enable_distinct_pushdown TO off;
-- Must pull even duplicate rows without pushdown. Verify that using EXPLAIN ANALYZE
EXPLAIN (ANALYZE, COSTS OFF, TIMING OFF, SUMMARY OFF) SELECT DISTINCT r1 FROM distinct_pushdown_table WHERE r1 <= 10;
                                                QUERY PLAN
-----------------------------------------------------------------------------------------------------------
 Unique (actual rows=1 loops=1)
   ->  Index Scan using distinct_pushdown_table_pkey on distinct_pushdown_table (actual rows=1000 loops=1)
         Index Cond: (r1 <= 10)
(3 rows)

-- Enable DISTINCT pushdown
SET yb_enable_distinct_pushdown TO on;
-- Must pull fewer rows with pushdown. Verify that using EXPLAIN ANALYZE
EXPLAIN (ANALYZE, COSTS OFF, TIMING OFF, SUMMARY OFF) SELECT DISTINCT r1 FROM distinct_pushdown_table WHERE r1 <= 10;
                                               QUERY PLAN
--------------------------------------------------------------------------------------------------------
 Unique (actual rows=1 loops=1)
   ->  Index Scan using distinct_pushdown_table_pkey on distinct_pushdown_table (actual rows=1 loops=1)
         Index Cond: (r1 <= 10)
(3 rows)

DROP TABLE distinct_pushdown_table;
